### DISCUSSION TITLE
Help with MPV on OSX15 Sequoia using unsupported MacBook Pro

### DISCUSSION THREAD

**Discussion by kickaxe60** [mpv-player/mpv#15388] [Q&A] [answered]
Curious to know if it's possible to get MPV working with my setup.  MPV played perfectly before my upgrade from Big Sur to Sequoia using Open Core Legacy Patcher.  After the upgrade, except for MPV, everything I threw at Sequoia works flawlessly (including VLC).  MPV starts and closes immediately.  I deleted the older version of MPV and installed 0.39 with no change.  VLC is great but I have so many options via the .config file with MPV so I would prefer to try to get it working.  Any help to get it working again with OSX15 would be appreciated.  

Below are some specs and an excerpt of the Crash report when I try to play an .mp4 video:

mpv 0.39.0 Copyright © 2000-2024 mpv/MPlayer/mplayer2 projects
libplacebo version: v7.349.0
FFmpeg version: 7.1
FFmpeg library versions:
   libavcodec      61.19.100
   libavdevice     61.3.100
   libavfilter     10.4.100
   libavformat     61.7.100
   libavutil       59.39.100
   libswresample   5.3.100
   libswscale      8.3.100

My Laptop:
Late 2013 15" MacBook Pro Retina upgraded with OCLP to OSX 15.1.1 Sequoia from OSX 11 Big Sur
2.6 GHz Quad-Core Intel Core i7
Intel Iris Pro 1536 MB
16 GB 1600 MHz DDR3
1TB SSD

Thread 16 Crashed:: vo
0   libsystem_kernel.dylib        	    0x7ff80b427b52 __pthread_kill + 10
1   libsystem_pthread.dylib       	    0x7ff80b461f85 pthread_kill + 262
2   libsystem_c.dylib             	    0x7ff80b382b19 abort + 126
3   libc++abi.dylib               	    0x7ff80b419163 abort_message + 258
4   libc++abi.dylib               	    0x7ff80b40a0ce demangling_terminate_handler() + 266
5   libobjc.A.dylib               	    0x7ff80b0a7d61 _objc_terminate() + 96
6   libc++abi.dylib               	    0x7ff80b41853b std::__terminate(void (*)()) + 6
7   libc++abi.dylib               	    0x7ff80b41b237 __cxxabiv1::failed_throw(__cxxabiv1::__cxa_exception*) + 27
8   libc++abi.dylib               	    0x7ff80b41b21c __cxa_throw + 69
9   libobjc.A.dylib               	    0x7ff80b09fccf objc_exception_throw + 317
10  CoreFoundation                	    0x7ff80b65f18f -[NSObject(NSObject) doesNotRecognizeSelector:] + 342
11  CoreFoundation                	    0x7ff80b5273f5 ___forwarding___ + 745
12  CoreFoundation                	    0x7ff80b527078 _CF_forwarding_prep_0 + 120
13  libMoltenVK.dylib             	       0x2dbc3dad5 std::__1::__function::__func<MVKGraphicsResourcesCommandEncoderState::encodeImpl(unsigned int)::$_6, std::__1::allocator<MVKGraphicsResourcesCommandEncoderState::encodeImpl(unsigned int)::$_6>, void (MVKCommandEncoder*, MVKMTLBufferBinding&)>::operator()(MVKCommandEncoder*&&, MVKMTLBufferBinding&) + 69
14  libMoltenVK.dylib             	       0x2dbc3a393 MVKGraphicsResourcesCommandEncoderState::encodeBindings(MVKShaderStage, char const*, bool, std::__1::function<void (MVKCommandEncoder*, MVKMTLBufferBinding&)>, std::__1::function<void (MVKCommandEncoder*, MVKMTLBufferBinding&, MVKArrayRef<unsigned int const>)>, std::__1::function<void (MVKCommandEncoder*, MVKMTLTextureBinding&)>, std::__1::function<void (MVKCommandEncoder*, MVKMTLSamplerStateBinding&)>) + 2147
15  libMoltenVK.dylib             	       0x2dbc3ad12 MVKGraphicsResourcesCommandEncoderState::encodeImpl(unsigned int) + 274
16  libMoltenVK.dylib             	       0x2dbbad65b MVKCommandEncoder::finalizeDrawState(MVKGraphicsStage) + 107
17  libMoltenVK.dylib             	       0x2dbba74d3 MVKCmdDraw::encode(MVKCommandEncoder*) + 579
18  libMoltenVK.dylib             	       0x2dbbab0f8 MVKCommandEncoder::encode(id<MTLCommandBuffer>, MVKCommandEncodingContext*) + 232
19  libMoltenVK.dylib             	       0x2dbbab564 MVKCommandBuffer::submit(MVKQueueCommandBufferSubmission*, MVKCommandEncodingContext*) + 244
20  libMoltenVK.dylib             	       0x2dbc13dc6 MVKQueueFullCommandBufferSubmission<1ul>::submitCommandBuffers() + 86
21  libMoltenVK.dylib             	       0x2dbc12197 MVKQueueCommandBufferSubmission::execute() + 263
22  libMoltenVK.dylib             	       0x2dbc0fe9a MVKQueue::submit(MVKQueueSubmission*) + 186
23  libMoltenVK.dylib             	       0x2dbc1000f VkResult MVKQueue::submit<VkSubmitInfo2>(unsigned int, VkSubmitInfo2 const*, VkFence_T*, MVKCommandUse) + 255
24  libMoltenVK.dylib             	       0x2dbb6e7ba vkQueueSubmit2 + 74
25  libplacebo.349.dylib          	       0x101302dfb vk_cmd_submit + 554
26  libplacebo.349.dylib          	       0x1013089f3 _end_cmd + 267
27  libplacebo.349.dylib          	       0x10131073e vk_pass_run + 2547
28  libplacebo.349.dylib          	       0x1012e092f pl_pass_run + 663
29  libplacebo.349.dylib          	       0x1012e433e pl_pass_run_vbo + 398
30  libplacebo.349.dylib          	       0x1012e092f pl_pass_run + 663
31  mpv                           	       0x100ed621c renderpass_run_pl + 616
32  mpv                           	       0x100eba44d gl_sc_dispatch_draw + 217
33  mpv                           	       0x100ec3fde finish_pass_fbo + 1036
34  mpv                           	       0x100ec3bab finish_pass_tex + 857
35  mpv                           	       0x100ebf26b pass_render_frame + 1028
36  mpv                           	       0x100ebe44b gl_video_render_frame + 1201
37  mpv                           	       0x100ecdce5 draw_frame + 83
38  mpv                           	       0x100eccbf3 vo_thread + 1084
39  libsystem_pthread.dylib       	    0x7ff80b462253 _pthread_start + 99
40  libsystem_pthread.dylib       	    0x7ff80b45dbef thread_start + 15

**Comment by Akemi** [comment, marked as answer]
you can try `vo=libmpv` instead of the new default `vo=gpu` + macvk context. it also probably works better on intel based macs.

your infos are a bit too vague to say anything and other infos are missing. this would be better placed as a proper issue instead of a discussion, since the issue templates asks for a lot of the needed infos.

this crash is in moltenvk. either the build is broken, metal isn't working properly on your end or moltenvk is broken.

**Comment by kickaxe60** [comment]
Hey Akemi, thanks so much for replying.  Your solution worked perfectly.  I am now able to play videos with mpv again!

Much appreciated.

**Reply by Akemi** [reply]
you are welcome.
